### DISCUSSION TITLE
What's the best way to get slate DOM elements based on cursor position?

### DISCUSSION THREAD

**Discussion by narkeeso** [ianstormtaylor/slate#4341] [Q&A]
I'm trying to build a medium.com like editor where a menu button pops up on the left when you insert a newline. The documentation doesn't seem to make it clear how to obtain refs to the nodes so you can render things around/near it.

**Comment by tiberiuichim** [comment]
We have an "inline floating toolbar" running with the following code: https://github.com/eea/volto-slate/blob/master/src/editor/ui/Toolbar.jsx#L37-L54

**Reply by narkeeso** [reply]
Thanks for the example. This looks similar to the example on the Slate site. Is there a more React way of selecting the components instead of using window.getSelection()?
